app/src/main/java/com/sdp/movemeet/view/profile/EditProfileActivity.java

```diff
@@ -151,15 +149,19 @@ public void onFailure(@NonNull Exception e) {
 
     @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE) // making this method always public for testing and private otherwise
     public void accessFirestoreUsersCollectionForUpdate() {
-        DocumentReference docRef = fStore.collection("users").document(userId);
-        Map<String, Object> edited = FirebaseInteraction.updateDataInFirebase(profileFullName, profileEmail, profilePhone, profileDescription);
-        docRef.update(edited).addOnSuccessListener(new OnSuccessListener<Void>() {
+        User user = new User(profileFullName.getText().toString(), profileEmail.getText().toString(), profilePhone.getText().toString(), profileDescription.getText().toString());
+        userManager.add(user, FirestoreUserManager.USERS_COLLECTION + "/" + userId).addOnSuccessListener(new OnSuccessListener() {
```

Comment:
If all of the calls to `userManager.add()` look like this,, you can probably re-write the UserManager's implementation of the add method so that it automatically formats the path with the `collection + '/' + id` format

Reply:
Yes, might be cleaner indeed! That's a very good remark, I will see what I can do!